dolphinscheduler-python/src/main/java/org/apache/dolphinscheduler/server/PythonGatewayServer.java

```diff
@@ -0,0 +1,193 @@
+//                    locations, timeout, tenantCode, taskRelationJson, taskDefinitionJson);
+//            return code;
+            return 1L;
+        } else {
+            // create process definition
+            Map<String, Object> result = processDefinitionService.createProcessDefinition(userPythonGateway, projectCode, name, description, globalParams,
+                    locations, timeout, tenantCode, taskRelationJson, taskDefinitionJson);
+            ProcessDefinition processDefinition = (ProcessDefinition) result.get(Constants.DATA_LIST);
+            return processDefinition.getCode();
+        }
+    }
+
+    public void execProcessInstance(long projectCode,
+                                    long processDefinitionCode,
+                                    String cronTime,
+                                    CommandType commandType,
+                                    FailureStrategy failureStrategy,
+                                    String startNodeList,
+                                    TaskDependType taskDependType,
+                                    WarningType warningType,
+                                    int warningGroupId,
+                                    RunMode runMode,
+                                    Priority processInstancePriority,
+                                    String workerGroup,
+                                    Long environmentCode,
+                                    Integer timeout,
+                                    Map<String, String> startParams,
+                                    Integer expectedParallelismNumber) {
+        executorService.execProcessInstance(userPythonGateway,
+                projectCode,
+                processDefinitionCode,
+                cronTime,
+                commandType,
+                failureStrategy,
+                startNodeList,
+                taskDependType,
+                warningType,
+                warningGroupId,
+                runMode,
+                processInstancePriority,
+                workerGroup,
+                environmentCode,
+                timeout, startParams,
+                expectedParallelismNumber);
+    }
+
+
+    // side object
+    public Map<String, Object> createProject(String name, String desc) {
+        return projectService.createProject(userPythonGateway, name, desc);
+    }
+
+    public Map<String, Object> createTenant(String tenantCode, String desc) throws Exception {
+        return tenantService.createTenant(userPythonGateway, tenantCode, queuePythonGateway.getId(), desc);
+    }
+
+    @PostConstruct
+    public void run() {
+        PythonGatewayServer app = new PythonGatewayServer();
+        // app is now the gateway.entry_point
+        // GatewayServer server = new GatewayServer(app,25334); to specific port
+        GatewayServer server = new GatewayServer(app);
```

Comment:
```suggestion
        GatewayServer server = new GatewayServer(this);
```

Hi @caishunfeng, next time when you try to add suggestions in PR, maybe you could directly [reviewing changes][1] like this comment. I make code reviewed more directly. what more, when PR author  accept your review,  you would become [co-author][2] after PR merge

[1]: https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/reviewing-proposed-changes-in-a-pull-request
[2]: https://docs.github.com/en/github/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors